dashboard/app/models/authentication_option.rb

```diff
@@ -35,6 +35,7 @@ class AuthenticationOption < ApplicationRecord
   validate :email_must_be_unique, :hashed_email_must_be_unique, unless: -> {UNTRUSTED_EMAIL_CREDENTIAL_TYPES.include? credential_type}
 
   validates :authentication_id, uniqueness: {scope: [:credential_type, :deleted_at], case_sensitive: true}
+  validates :authentication_id, if: :lti?, format: {with: /\A(\S+\|\S+\|\S+)\z/, message: "For LTI authentication_options, format must be 'issuer|audience|subject'"}
```

Comment:
Someone should double check my sketchy regex skills here haha

Reply:
Maybe `split` the string on `|` instead and verify the 3 strings in the resulting array are all alphanumeric? Ruby’s string functions are so user-friendly that I find it’s rare I need to use regex.

Reply:
I could do this, I have noticed other places in our codebase using regex, which allows you to use the `format:` method. Is there a preference to move away from regex? I personally don't like regex, but was trying to standardize on what looked like the preferred way.

Reply:
Sorry for the very slow response. I don't think we have ever formalized a code style about whether/when to use regex. They are very hard to read, and I think it's worth using Ruby's wonderful string methods and not resort to regex unless there isn't a more intuitive and readable way to accomplish it.